### ISSUE TITLE
Report Module

### ISSUE THREAD

**Issue by Gyukat** [EQAR/deqar_frontend#3]
https://github.com/EQAR/eqar_frontend/issues/136?_pjax=%23js-repo-pjax-container

**Comment by Gyukat** [comment]
TESTING MANUAL REPORT SUBMISSION:

**Institution / Programme Assignment:**

- Main Form
  -  **FIXED** I still find the institution and programme assignment and listing a bit confusing…maybe the label on Institution should read “Add Institution” and the “Add Programme” button could come directly below it when applicable.  Then the assigned institutions and programmes would display below **(I might slightly indent these as well to visually distinguish them.)** _Note: Assigned List and it's button belongs to one component, so I can't place it above institutions, only above it's respective ListBox_ 
      - **OK..but can you still indent the "assigned" lists?**
  - **FIXED** When you enter an institution or programme, the display of assigned institutions and programmes cuts the words vertically.
  - There should be a functioning link (eventually) from the assigned institution to the correct institution record.  (This works REALLY WELL for programmes…bravo)
  - **FIXED** _THX_ For the assigned programmes, like for the assigned files, there should be a concatenated string entry combining the programme name and degree (if present) with QF level in parenthesis.
- Pop Up
  - **FIXED** _THX_ Btw, short cycle should come before first cycle in the drop down lists…it should not be alphabetised but in progressive order.

**File Assignment:**

- Main Form
  - **FIXED** move up Add File button above assigned files (like the logic above for programmes) 
      - **OK..but can you still indent the "assigned" lists?**
  - **FIXED** _THX_ the assigned file entry should contain information from several fields (file name and languages in parentheses OR file URL and languages in parentheses).  It can wrap.
- Pop-Up
  - long file names still go off the end of the pop up box. 
      - **Uploaded screen shot below!**
  - if you browse for a file, and then want to cancel the option, it is not possible.
  - **FIXED (see below)** _THX_ if you enter files as both URL and BROWSED, and then want to cancel the URL, the “x” doesn’t work.

**Other:**

- **FIXED** _THX_ If you enter a badly formatted agency website URL , and then want to cancel the URL, the “x” doesn’t work. _Note: This shouldn't even be there, was some weird thing from the Bootstrap CSS framework. They never implemented, that it should empty the field. :) Removed._

**Submission:**

- not sure what should happen when you successfully submit, but intuitively I would expect to be taken back to the list and to SEE my new report in the list.  Then I would happily do the next entry.
(Currently, you have some very nice messages, which show both the record being created and the file being uploaded.  These should definitely STAY.  BUT after this, you are presented with empty record.)
- **FIXED** _THX_ submit button on right side, no?
- **FIXED** submission process weird in one case…I put a too early validity date… a very nice red message came up at the top of the record to tell me that this is not good.  _(Though we might consider using the same kind of field highlighting with message underneath in this case.)_ After this, the submission process seems to be stuck…I was able to correct the date and cancel the message, but the button is still spinning.  Had to start the whole process again. **Note - I fixed in the quick way, so now you're getting back the control of the button (I forgot to reset the spinner if the request failed.) / I'll include the date validation vs agency date validation on the form directly, so the error message could be much more proper and design-wise the same as the others (under the field in red)**
    - **I awaiting the final fix...right now, it works again, but the red alert still hangs suspended after the value is fixed and record accepted.** 
- Maybe I crashed the system, because it is spinning again…both times with fake EKKA record.

**Comment by Gyukat** [comment]
TESTING REPORT RECORD VIEW:

**Visual Problems:**

- **FIXED (let me know if it's acceptable this way)** Make fields white background and field boundaries light or invisible. _My personal note here is, that I agree with making the background a bit lighter, but you have to make it clear, that it's a non-editable field on a form. / With simple text it will look ugly as hell_ 
- **FIXED** Get rid of drop-down arrows and “x” on some values in pop-ups. These should not just be deactivated but not visible.
- **FIXED** when there is no valid to, then leave blank…don’t leave the “YYYY-MM-DD” in the box.

**Other:**

- **FIXED** (See notes for concatenated strings on the Assigned files and programmes entries from above…these apply here too.)
- **FIXED** Again, I would have the Edit button available from the View Report Record and not from the Main Menu…let’s assume that most people will look at the record, and then decide to edit.
- **FIXED** Edit button should be on right side.  Hide/Show can go near or somewhere else (nice touch!).

**Comment by JoshBone** [comment]
>   - if you browse for a file, and then want to cancel the option, it is not possible.

Currently it's working in a way, that you have to select browse and then press cancel. Should I include a cancel button somewhere? 

- **Definitely...the cancel does not get rid of an existing value as I can see...so we need some sort of delete or erase function.**

But just to clarify, you want to cancel, if you accidentally selected a file and you rather want to enter a URL. Because otherwise, you should press X on the main form to remove the file and save the record.

**YES...TRUE**

**Comment by JoshBone** [comment]
> long file names still go off the end of the pop up box.

Can you post a screenshot?

**Comment by JoshBone** [comment]
> not sure what should happen when you successfully submit, but intuitively I would expect to be taken back to the list and to SEE my new report in the list. Then I would happily do the next entry (Currently, you have some very nice messages, which show both the record being created and the file being uploaded. These should definitely STAY. BUT after this, you are presented with empty record.)

Should we jump to the list page, or rather show the newly created record (with the flags and dates) in 'view' mode?

**Comment by Gyukat** [comment]
> > long file names still go off the end of the pop up box.
> 
> Can you post a screenshot?

<img width="539" alt="screen shot 2019-03-03 at 20 00 44" src="https://user-images.githubusercontent.com/33084539/53700089-16c50700-3def-11e9-9fb0-6a13ef52b49e.png">

**Comment by Gyukat** [comment]
> > not sure what should happen when you successfully submit, but intuitively I would expect to be taken back to the list and to SEE my new report in the list. Then I would happily do the next entry (Currently, you have some very nice messages, which show both the record being created and the file being uploaded. These should definitely STAY. BUT after this, you are presented with empty record.)
> 
> Should we jump to the list page, or rather show the newly created record (with the flags and dates) in 'view' mode?

I guess that I would be open to either.  It would be nice to add a "New record" button on either the record or the menu...otherwise, are we depending on the user to use the side menu?

**MOVE BUTTON TO THE RIGHT SIDE**

**Comment by Gyukat** [comment]
**NEW THINGS:**
SIDE MENU
- Why did you combine the uploaded reports and submission under My Reports?  I preferred it separate.

MENU (My Uploaded Reports)
- What is the difference between the Menu under My Agencies and My Uploaded Reports?  I don't think the first has much to do with Agencies and there is a neat according feature on the first that I don't see on the second.
- **FIXED(?)** The institution programme name should be underlined to show that it is a link---as you do in the view/edit forms.  I would make this column substantially wider than the others as it is really the MAIN column. - _This one should be okay, I'm not sure why you didn't see it. Always press refresh when you try the system first._
- Ugh, I think we somehow need to show the Agency on the My Uploaded Reports list...for those few cases when one agency has the proxy for others.  Is there some way to do this without messing up the whole thing?
- **FIXED** Multiple country names in the list need a space between.
- I think the date filters can be by year only.  This will save space and be more intuitive.  You can put YYYY in the filter boxes to clarify.  The uploaded date column can be narrower, by either removing the time or by putting it on the second line.
- Flagging column can be narrower by changing drop down to none, low, high.  The flags themselves can also read "low" and "high".  The column can be labelled Flag Level.
- **_This is a bigger deal, because currently the react-table component doesn't support this feature, we should manually implement it. I'll keep it on my schedule, but it's not a priority at the moment._** At bottom full number of reports on the list should be stated outright like "Reports 1-10 of 265" or "1-10 of 265 reports". This is better than just the number of pages, which requires that users do a maths problem to know how many reports.
- **FIXED** Consider putting a button for "Submit New Report" at the bottom right.

**Comment by Gyukat** [comment]
REPORT RECORD

GENERAL
- Header
  - **FIXED (it's super ugly in my opinion :) )** Indent all three assigned fields (not just the values but the whole field with labels...you can use more space if you are worried it will look weird.
    - **Discussed with token user (i.e. Gyuri) and his input is: 1) get rid of "Assigned ..." labels...they are superfluous and confusing with the "Add..." buttons and fields...we can simply array the lists under the Add buttons and fields; alternatively, you can just put “Institutions”, “Programmes” and “Files” instead of “Assigned…“; 2) indent more, maybe aligning with the beginning of the text under Flags at the bottom; 3) Add Programme and Add File buttons should not be indented, just the assigned values.**
  - **_This can be fixed in the database, please review the values with Colin._** Multilanguages like Dutch; Flemish look really weird in a list.  The problem is the the ";" is actually a stronger divider than the ",".  Thus, Dutch; Flemish, English looks really weird in both the Administrative and Public UI.  Worth thinking about changing the entries to "Dutch/Flemish" so that the list would say "Dutch/Flemish, English"
- Footer
  - **FIXED** Record Created/Record Updated should probably include the "by whom" and not just the when and what.  Sorry, I forgot that on the design.
  - **FIXED** Record created and Record updated should be Record History.  It should include the When and by Whom.  
  - There should also be a chance when saving the edited record to attach a "comment"  to the entry.
  - We are missing internal notes both view and enter.

VIEW
- The two pop-up boxes (i.e. programmes and files) have both an "x" and a "close" button.  We should decide which and be consistent.
- **FIXED** I still don't like the field boundaries.  It looks like you should enter something.  Are you worried about the aesthetics if we get rid of the boundaries?  Is there no way to make it look nice and also simply be "flat" data on a white background. (in the very least make the field background pure white.)

CREATE
- The red alerts stay around the field after you have already fixed them...shouldn't they disappear when you enter the field and put in a new value. 
- Again, it is very strange to end up nowhere after you submit...otherwise, looking better...
- When you use the browse, we still get the "file location was not provided" note and yellow flag on the notification but not in the system.

EDIT
- **_After I'm back from SA_** I think we should consider putting a place for Agencies or Admin to signal a problem with a record.  This could be a non-assigned low-level flag, but it could be an alert with a mandatory note attached.  Don't kill me...it just seems like the missing piece.  Right now. Flags are given automatically and can be removed by the system (I guess) and/or overridden by the Admin.  Admin can also enter an internal note which can be viewed by anyone seeing the record.  This can explain any overrides or anomalies.  Still, we don't have a way for people to manually signal a problem, whether the agency or the administrator.   

ADMIN EDIT
- Does this exist?  Where can I add internal notes/comments?  Where can I deactivate flags?

**Comment by JoshBone** [comment]
> Why did you combine the uploaded reports and submission under My Reports? I preferred it separate.

I thought that it looks a bit more organised, but define the menu how you would like to see, and that's how it's going to happen...

**Comment by JoshBone** [comment]
> What is the difference between the Menu under My Agencies and My Uploaded Reports? I don't think the first has much to do with Agencies and there is a neat according feature on the first that I don't see on the second.

I guess My Agencies will have the form where you can actually edit the information about agencies that belonging to your user.

**Comment by JoshBone** [comment]
> I think the date filters can be by year only. This will save space and be more intuitive. You can put YYYY in the filter boxes to clarify. The uploaded date column can be narrower, by either removing the time or by putting it on the second line.

You mean by removing the active checkbox? - I've spent quite some time to implement it... :)

**Comment by JoshBone** [comment]
> I still don't like the field boundaries. It looks like you should enter something. Are you worried about the aesthetics if we get rid of the boundaries? Is there no way to make it look nice and also simply be "flat" data on a white background. (in the very least make the field background pure white.)

The thing is that the data is the same form-data. I think it looks much nicer that you can change the view when you edit, and you just have to enter the data into real fields... I can make all kinds of styles, lighter border, lighter text color, but still want to keep them as fields.

**Comment by JoshBone** [comment]
> When you use the browse, we still get the "file location was not provided" note and yellow flag on the notification but not in the system.

This won't be the case, when we have the email digesting method. (After I come back from SA)

**Comment by JoshBone** [comment]
> Does this exist? Where can I add internal notes/comments? Where can I deactivate flags?

Not yet.
Should this be some separate admin menu, or rather the same report list (Reference Data -> Reports) should have editing options if the user has admin privileges?

**Comment by JoshBone** [comment]
> Flagging column can be narrower by changing drop down to none, low, high. The flags themselves can also read "low" and "high". The column can be labelled Flag Level.

Can I change this system (frontend + backend) wise? So that it will affect everything? (Outgoing email, and stuff)

**Comment by Gyukat** [comment]
> > What is the difference between the Menu under My Agencies and My Uploaded Reports? I don't think the first has much to do with Agencies and there is a neat according feature on the first that I don't see on the second.
> 
> I guess My Agencies will have the form where you can actually edit the information about agencies that belonging to your user.

But the accordion feature is super neat...can't we use it in My Uploaded Reports and Reference Reports or is it too "busy"?

**Comment by Gyukat** [comment]
> > I think the date filters can be by year only. This will save space and be more intuitive. You can put YYYY in the filter boxes to clarify. The uploaded date column can be narrower, by either removing the time or by putting it on the second line.
> 
> You mean by removing the active checkbox? - I've spent quite some time to implement it... :)

No, keep the checkbox...it is cool.  I mean that in the field itself allow the year.  It should be enough to filter and is cleaner and takes less space.

**Comment by Gyukat** [comment]
> > I still don't like the field boundaries. It looks like you should enter something. Are you worried about the aesthetics if we get rid of the boundaries? Is there no way to make it look nice and also simply be "flat" data on a white background. (in the very least make the field background pure white.)
> 
> The thing is that the data is the same form-data. I think it looks much nicer that you can change the view when you edit, and you just have to enter the data into real fields... I can make all kinds of styles, lighter border, lighter text color, but still want to keep them as fields.

Well, then, make the background white and quite light or checked borders.

**Comment by Gyukat** [comment]
> > Does this exist? Where can I add internal notes/comments? Where can I deactivate flags?
> 
> Not yet.
> Should this be some separate admin menu, or rather the same report list (Reference Data -> Reports) should have editing options if the user has admin privileges?

I was thinking the latter...any opinions?

**Comment by Gyukat** [comment]
> > Flagging column can be narrower by changing drop down to none, low, high. The flags themselves can also read "low" and "high". The column can be labelled Flag Level.
> 
> Can I change this system (frontend + backend) wise? So that it will affect everything? (Outgoing email, and stuff)

I would think it would be fine, as long as the labels are changed to read "flag level" in every instance.

**IN FACT JUST SAY "FLAG" IN THE LABEL AND THEN USE SIMPLY: HIGH, LOW, NONE.  IT IS MUCH SIMPLER!!!**

**Comment by Gyukat** [comment]
> > Why did you combine the uploaded reports and submission under My Reports? I preferred it separate.
> 
> I thought that it looks a bit more organised, but define the menu how you would like to see, and that's how it's going to happen...

I like: 

MY DATA

- My Profile
- My Agences
- My Reports
- Flags and Alerts

MENU

- Submit Reports
  - Manual Submission
  - Upload CSV 
- Reference Data
  - Agencies 
  - Reports
  - Institutions
  - Countries
- Statistics
- Documentation

**Comment by JoshBone** [comment]
> > > I think the date filters can be by year only. This will save space and be more intuitive. You can put YYYY in the filter boxes to clarify. The uploaded date column can be narrower, by either removing the time or by putting it on the second line.
> > 
> > 
> > You mean by removing the active checkbox? - I've spent quite some time to implement it... :)
> 
> No, keep the checkbox...it is cool. I mean that in the field itself allow the year. It should be enough to filter and is cleaner and takes less space.

It's acting like that already. :)

**AH, IT WASN'T CLEAR WHEN I TRIED IT...THEN YOU CAN DEFINITELY MAKE THE COLUMN MORE NARROW FOR BOTH DATE FIELDS.  PUT THE TIME ON SECOND LINE IN THE UPLOADED COLUMN.**

**Comment by JoshBone** [comment]
> > > Why did you combine the uploaded reports and submission under My Reports? I preferred it separate.
> > 
> > 
> > I thought that it looks a bit more organised, but define the menu how you would like to see, and that's how it's going to happen...
> 
> I like:
> 
> MY DATA
> 
> * My Profile
> * My Agences
> * My Reports
> * Flags and Alerts
> 
> MENU
> 
> * Submit Reports
>   
>   * Manual Submission
>   * Upload CSV
> * Reference Data
>   
>   * Agencies
>   * Reports
>   * Institutions
>   * Countries
> * Statistics
> * Documentation

**FIXED**


_**AGREE** KM_ Add link to documentation, no?  Deactivate all the non-active menu items (e.g. My Agencies).

**Comment by JoshBone** [comment]
> > > I still don't like the field boundaries. It looks like you should enter something. Are you worried about the aesthetics if we get rid of the boundaries? Is there no way to make it look nice and also simply be "flat" data on a white background. (in the very least make the field background pure white.)
> > 
> > 
> > The thing is that the data is the same form-data. I think it looks much nicer that you can change the view when you edit, and you just have to enter the data into real fields... I can make all kinds of styles, lighter border, lighter text color, but still want to keep them as fields.
> 
> Well, then, make the background white and quite light or checked borders.

**FIXED / I think it looks cool, like this**

_**AGREE** KM_

**Comment by Gyukat** [comment]
> > > > I still don't like the field boundaries. It looks like you should enter something. Are you worried about the aesthetics if we get rid of the boundaries? Is there no way to make it look nice and also simply be "flat" data on a white background. (in the very least make the field background pure white.)
> > > 
> > > 
> > > The thing is that the data is the same form-data. I think it looks much nicer that you can change the view when you edit, and you just have to enter the data into real fields... I can make all kinds of styles, lighter border, lighter text color, but still want to keep them as fields.
> > 
> > 
> > Well, then, make the background white and quite light or checked borders.
> 
> **FIXED / I think it looks cool, like this**
> 
> _**AGREE** KM_

YOU CAN USE THE SAME FORMAT FOR THE INSTITUTION VIEW PAGE.  WE WON'T YET UPDATE, BUT LET'S MAKE THE "LOOK" THE SAME.

**Comment by Gyukat** [comment]
THINGS LEFT TO DO

TODAY:
New Report / Edit

- **FIXED** Assigning Institutions, Programmes and Files.  So, we have spoken with Colin and have one final formatting proposal for this.  I can easily tell you in person.
- Enable the link to the Institution record view.
- Pop-Up:  long file names still go off the end of the pop up box.
- Pop-Up:  if you browse for a file, and then want to cancel the option, it is not possible.
- not sure what should happen when you successfully submit, but intuitively I would expect to be taken back to the list and to SEE my new report in the list. Then I would happily do the next entry.
(Currently, you have some very nice messages, which show both the record being created and the file being uploaded. These should definitely STAY. BUT after this, you are presented with empty record.)
- **FIXED** When there are two institutions, then there should also be two URLS linked from the admin part of the report record.

**Comment by Gyukat** [comment]
**SIDE MENU:**

- **FIXED** - Get rid of sub-menus, use accordion style for My Data
- **FIXED** - Dashboard should show My Reports Menu but include stats at the top; include sort but not filters for now.
- **WILL DO AFTER SA / Please define the fields needed** - My Reports Menu should include an accordion feature, Colin and Katie will specify the fields...
- **FIXED** My Agencies use the current information with new look

MY REPORTS MENU

- **PARTIALLY FIXED** Date Fields: Condense upload date data (either smaller font or two lines); Date should be labelled  "Validity" and the Active checkbox should be after the date filter field and can use the icon---should be checked automatically and logic reversed; both filter boxes should be made smaller and include "YYYY" and only allow four characters
- **WILL DO AFTER SA** Flags: should become: high, low, none.   Label can remain flags.
- **FIXED** Activity filter should work; you can keep it by type.
- **FIXED FOR NOW** Eventually drill down filters; for now, shorten the filter lists to only those things in the system.
- **FIXED** Three filter values should not have "x" for deletion and should be shortened to fit in the field.
- **FIXED** Add DEQAR ID field.
- **FIXED** Submit New Report button on right side
- Eventually the number of records should be shown at bottom.
- **FIXED** Make same change to general Reports menu
- **FIXED** Make Institution menu like Reports menu: underline/linked institution name to open view record. _Alternative names are not displayed correctly_

New Report/Edit Report

- **FIXED** Change Add Institutions, Programmes and Files as discussed
- **WILL DO AFTER SA** Enable the link to the institution view (pop-up)
- Fix the File Upload feature (long names and cancelling)
- Upon Submission New Record returns to My Reports menu and Edited Record returns to View Record

Documentation:
- Add links to the top.

**Comment by Gyukat** [comment]
**Report Submission Form**

 - **FIXED** When you enter an institution or programme, the display of assigned institutions and programmes cuts the words vertically. _(still need to check at home)_
 - ~long file names still go off the end of the pop up box.~

**Report View Form**

 - **FIXED** Get rid of “x” on the multi-value country/language options in the pop-up boxes in view mode. 
 - **FIXED** The two pop-up boxes (i.e. programmes and files) have both an "x" and a "close" button. We should decide which and be consistent.

**My Reports Menu**

- Missing "Agency" information for proxy submitters...we should find a way to do this without messing up the whole menu for the average user.
- **NEW!** When I change the row count at the bottom, the final list seems to be a different length...so 13 pages for 10; 6 pages for 20; 5 pages for 25; but only 2 pages for 50 and 1 page for 100.   Still have no idea how many results there actually are but I would assume something like 120-130?
- **FIXED** Multiple country names in the list need a space between and should wrap.  You could also start each on a new line.
- Change flagging label to : "low", "high" and "none"

**Reports Menu**

- **FIXED** The list of countries is not the same as for My Reports.  It should be shorter like the MyReports list.

**Comment by Gyukat** [comment]
Don't forget the list above!!!!

**Comment by JoshBone** [comment]
> The two pop-up boxes (i.e. programmes and files) have both an "x" and a "close" button. We should decide which and be consistent.

What's the decision with this one?

**Comment by Gyukat** [comment]
**FIXED** The best solution would be to use grey "close" buttons at the bottom left corner of all the pop up boxes consistently in all cases (i.e. create and edit as well as view).  Then for the View you just keep the button there.  

If this is too hard and you have to choose, then just the "x" because it is already used on the create and edit...

**Comment by JoshBone** [comment]
Regarding to the previous one:
Modals can be closed via 'Esc' button and also when you click outside the box. 
If I got rid of the 'X' button and have only the grey 'Close' should I get rid of these extra two behaviour as well?

**Comment by Gyukat** [comment]
I can't think of any particular problem if you keep the existing methods (all three).  Am I missing something?  Is there reason, we would need to click to update data or something?

**Comment by Gyukat** [comment]
Josh [9:54 AM]
Hi @Colin and @Gyukat,
Couple of things which I did in the last couple of days (which are ready to be tested on admin.test.deqar.eu):
1. Users with ‘admin’ rights are able to edit the records in the reports menu as well. (The edit button is not displayed for regular users and obviously the endpoint won’t accept edit requests.) I have a question here. So django has a built in way to mark admin users. Those are the users who can log-in to DjangoSuit (who has is staff status) are considered as admin. I saw in DjangoSuit that several agencies have this option set. Is there any particular reason for this, or it was ad-hoc/accidental? If there is a reason, I can include Groups and we can create a separate Group entity for EQAR staff and run the checks against that group.
2. When ‘Submit’ is pressed on the Edit Report form, the record won’t be send immediately, but an optional input field appears where you can add some detail about your updates.
2. A) Minor change: I’ve switched InfoBox (flags and update log) and Submit / Close buttons and made the Info hidden in general. I thought it puts the managing buttons more in focus. Are you okay with this?
2. B) Each update is now creating a separate log entry (ReportUpdateLog table) *only when the update detail was filled*. I thought that otherwise accidental saves would always add a new line, so the update log would be huuuge… We can still decide to do so, but then let’s just only display the latest 3 - 5 updates.
2. C) _Improvement for later_ - I wanted to use the same kind of popup for submission, but I had some shortage with the current form library we are using, namely with the current version, you can’t fire validation separately, only together with submission. The latest/newest version can do this trick, but for some reason it breaks the already functioning popup-form functionality. So it needs further investigation which I’d like to do later, not to be held up by this.
3. Added the function ‘Request Deletion’. After a confirmation window a ‘high level’ flag (with the text: ’Deletion Requested.’) will be added to the report.
3. A) _Improvement for later_ - I’d like to hide the Request Deletion button once a record was already “deleted”. (Currently new requests aren’t really doing anything…)
4. Added the function for ‘admin’ to be able to remove flags. This can be achieved in Edit mode only with the regular ‘x’ next to the flag. After confirmation the request for removing a flag is immediately sent to the endpoint, so there is no need to submit the report once again… What happens in the background is that in the flag table active will be set to ‘False’ and ‘removed_by_eqar’ will be set to True. This way we can avoid the flag to be triggered once again if flags are re-assigned automatically. An updateLog will be also created. Question: ‘Should I include the flag text to the update log?’
Let me know what you guys think any text that should be updated, etc...
Institution Form is on the way I’m really hoping to roll out that one this week.

Colin [3:45 PM]
Quick response re. 1) I think that was accidental, as agencies shouldn't log in to DjangoSuit. So in my view we can use the staff status to signal admin/full edit rights.

Gyukat [4:41 PM]
Agree…1) accidental.  3a) don’t hide, but deactivate…so that people can see that it was already requested.

**Comment by Gyukat** [comment]
> Josh [9:54 AM]
> Hi @colin and @Gyukat,
> Couple of things which I did in the last couple of days (which are ready to be tested on admin.test.deqar.eu):
> 
>     1. Users with ‘admin’ rights are able to edit the records in the reports menu as well. (The edit button is not displayed for regular users and obviously the endpoint won’t accept edit requests.) I have a question here. So django has a built in way to mark admin users. Those are the users who can log-in to DjangoSuit (who has is staff status) are considered as admin. I saw in DjangoSuit that several agencies have this option set. Is there any particular reason for this, or it was ad-hoc/accidental? If there is a reason, I can include Groups and we can create a separate Group entity for EQAR staff and run the checks against that group.

SO, YES, THE EXISTING SETTING SHOULD BE FINE.  WE NEED TO SIMPLY CHANGE THE MISTAKEN AGENCY STATUS.
> 
>     2. When ‘Submit’ is pressed on the Edit Report form, the record won’t be send immediately, but an optional input field appears where you can add some detail about your updates.

THIS IS VERY NICELY DONE!  I REALLY LIKE THE LOOK OF IT.  AND VERY EASY FOR END USERS. 

ONLY CRITIQUE IS THAT THE SUBMIT BUTTON TAKES YOU BACK TO THE EDIT RECORD.  WHEN YOU SUBMIT AN EDITED RECORD, IT SHOULD TAKE YOU TO THE RECORD IN "VIEW MODE" SO YOU CAN SEE YOUR CHANGES AND ALSO HOW THEY ARE REFLECTED IN THE INFO.  
> 
>     3. A) Minor change: I’ve switched InfoBox (flags and update log) and Submit / Close buttons and made the Info hidden in general. I thought it puts the managing buttons more in focus. Are you okay with this?
> 
THIS IS FINE AS LONG AS THE "CLOSE BUTTON" IS CONSISTENTLY ON THE LEFT SIDE THROUGHOUT THE SYSTEM.

>     4. B) Each update is now creating a separate log entry (ReportUpdateLog table) _only when the update detail was filled_. I thought that otherwise accidental saves would always add a new line, so the update log would be huuuge… We can still decide to do so, but then let’s just only display the latest 3 - 5 updates.
> 
I THINK WE SHOULD INCLUDE ALL, SHOW 5 AND INCLUDE A "SEE MORE"...I HOPE THAT THERE WON'T BE TOO MANY CASES OF BLUNDERS.  LET'S ASSUME THAT THE AVERAGE PERSON WON'T REALLY TOUCH THEIR RECORDS VERY OFTEN.  

>     5. C) _Improvement for later_ - I wanted to use the same kind of popup for submission, but I had some shortage with the current form library we are using, namely with the current version, you can’t fire validation separately, only together with submission. The latest/newest version can do this trick, but for some reason it breaks the already functioning popup-form functionality. So it needs further investigation which I’d like to do later, not to be held up by this.
> 
FINE.  WE STILL HAVE THE PROBLEM THAT THERE IS AN ALERT WHEN PEOPLE USE THE UPLOAD FILE METHOD.

>     6. Added the function ‘Request Deletion’. After a confirmation window a ‘high level’ flag (with the text: ’Deletion Requested.’) will be added to the report.

THIS IS GREAT FOR END USERS.  THE ADMIN SHOULD ACTUALLY HAVE A CORRESPONDING: "ACCEPT DELETION/DELETION" ...AND BOTH THIS BUTTON AND THE FLAG REMOVAL "BUTTON" SHOULD INCLUDE A POSSIBILITY TO PROVIDE A NOTE TO INFORM END USERS WHY THE REQUEST WAS ACCEPTED OR REJECTED.  (PROBABLY SHOULD BE HOOKED TO AN ALERT AT SOME POINT...)
> 
>     7. A) _Improvement for later_ - I’d like to hide the Request Deletion button once a record was already “deleted”. (Currently new requests aren’t really doing anything…)

YOU CAN DEACTIVATE THE BUTTON--PERHAPS INCLUDE A MOUSEOVER THAT DELETION HAS BEEN REQUESTED...I THINK THAT YOU MIGHT CONSIDER SOME MORE VISUAL SIGNAL THAT A RECORD HAS BEEN REQUESTED FOR DELETION...MAYBE IT COULD BECOME LIGHT RED AND LIST "UNDER CONSIDERATION FOR DELETION".  RIGHT NOW, IT IS REALLY HARD TO TELL THAT SOMEONE HAS REQUESTED DELETION...YOU HAVE TO OPEN UP THE INFO.
> 
>     8. Added the function for ‘admin’ to be able to remove flags. This can be achieved in Edit mode only with the regular ‘x’ next to the flag. After confirmation the request for removing a flag is immediately sent to the endpoint, so there is no need to submit the report once again… What happens in the background is that in the flag table active will be set to ‘False’ and ‘removed_by_eqar’ will be set to True. This way we can avoid the flag to be triggered once again if flags are re-assigned automatically. An update Log will be also created. Question: ‘Should I include the flag text to the update log?’
>        Let me know what you guys think any text that should be updated, etc...
>        Institution Form is on the way I’m really hoping to roll out that one this week.
> 
YES, THIS LOOKS NICE.  AS I MENTIONED ABOVE, IT SHOULD INCLUDE AN OPTION TO PROVIDE A NOTE.  I THINK THE REMOVED FLAG TEXT CAN BE ON THE LOG AS WELL AS ANY NOTE PROVIDED...AM I MISSING SOMETHING?

I HAVEN'T NOTICED ANY PARTICULAR PROBLEMS WITH TEXT...BUT I WILL KEEP LOOKING.
> 
> Colin [3:45 PM]
> Quick response re. 1) I think that was accidental, as agencies shouldn't log in to DjangoSuit. So in my view we can use the staff status to signal admin/full edit rights.
> 
> Gyukat [4:41 PM]
> Agree…1) accidental. 3a) don’t hide, but deactivate…so that people can see that it was already requested.

**Comment by Gyukat** [comment]
BTW, I TRIED TO UPDATE A RECORD BY ADDING A URL.  IT SEEMED TO ACCEPT BUT THE DATA DID NOT SHOW UP.

**Comment by Gyukat** [comment]
JOSH, I KNOW YOU ARE GOING TO HATE ME FOR THIS...BUT CAN'T THE ADMIN USERS HAVE ANOTHER SECTION OF THE MENU TO MANAGE FLAGGED RECORDS.  YOU CAN DO THIS IN LATE JULY...AFTER OTHER PARTS ARE FINISHED.

THIS WOULD MEAN THAT THERE WOULD BE A REPORT LIST AND INSTITUTION LIST AND THESE WOULD HAVE ONLY FLAGGED RECORDS.  THE MENU FIELDS COULD BE ADJUSTED TO FIT THE TASK OF MANAGING FLAGS.  WHEN WAS FLAG ADDED, BY WHOM, FLAG LEVEL/FLAG TYPE/COMMENT.  I THINK/HOPE IT WOULD BE EASY AS YOU COULD JUST USE THE SAME BEAUTIFUL MENU STRUCTURE THAT YOU HAVE MADE (THIS IS ME ATTEMPTING TO BUTTER YOU UP!).

**Comment by Gyukat** [comment]
What happens when a bad link is provided?

Look at this:  http://www.pka.edu.pl/raporty/2014/11/14/2014._UMed_WWL_oc._inst.pdf 

<img width="933" alt="Screen Shot 2019-08-19 at 20 38 39" src="https://user-images.githubusercontent.com/33084539/63290235-51e5c480-c2c1-11e9-955e-ef2344b0882f.png">

It is the second "institutional evaluation" on this page...https://www.eqar.eu/qa-results/search/institution/?id=1970

dead link.

**Comment by Gyukat** [comment]
no…it is fine
